src/libraries/System.Private.CoreLib/src/System/ArgumentException.cs

```diff
@@ -97,5 +99,19 @@ private void SetMessageField()
         }
 
         public virtual string? ParamName => _paramName;
+
+        /// <summary>Throws an <see cref="ArgumentNullException"/> if <paramref name="argument"/> is null. Throws an <see cref="ArgumentException"/> if <paramref name="argument"/> is an empty string.</summary>
+        /// <param name="argument">The reference type argument to validate as non-null and non-empty.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        public static void ThrowIfNullOrEmpty([NotNull] string? argument, [CallerArgumentExpression("argument")] string? paramName = null)
```

Comment:
You will also need to add this to the System.Runtime.cs reference assembly file.

Reply:
And please also add tests in System.Runtime\tests.  You can use ArgumentNullException.ThrowIfNull as an example:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/tests/System/ArgumentNullExceptionTests.cs#L52